README.md

```diff
@@ -1,44 +1,95 @@
 # IntelITT.jl
 
-## Basic usage
+*Julia interface for the Intel Instrumentation and Tracing APIs*
 
-We recommend to use "Start Paused" mode in conjuction with `IntelITT.resume()` and `IntelITT.pause()` to measure a specific instrumented part of your code (and not, e.g., julia startup and compilation).
+IntelITT.jl provides a high-level Julia interface to the [Intel Instrumentation and Tracing
+Technology APIs](https://github.com/intel/ittapi) APIs. These make it possible to control
+and enrich the collection of trace data when profiling Julia code with Intel tools such as
+VTune.
+
+When importing IntelITT.jl, a library that provides stub implementations of the IntelITT
+APIs will be loaded. In the case your application is being profiled, these stubs will be
+replaced by an actual implementation. This means that you should always be able to use
+IntelITT.jl without having to worry about whether you are being profiled or not.
+
+
+## Collection control
+
+When actually using Intel tools to profile your application, it's recommended to start the
+profiler in paused mode, and use IntelITT.jl to only instrument the parts of your code that
+you are interested in. This can be done as follows:
 
 ```julia
 using IntelITT
 
-# Check if we are running under VTune
-@assert IntelITT.available()
-
-function profile_test(n)
-    for i = 1:n
-        A = randn(100,100,20)
-        m = maximum(A)
-        Am = mapslices(sum, A; dims=2)
-        B = A[:,:,5]
-        Bsort = mapslices(sort, B; dims=1)
-        b = rand(100)
-        C = B.*b
-    end
-end
-
-# Compile code once
-profile_test(1)
+# warm-up, compile code, etc
 
-# Resume profiling
 IntelITT.resume()
-profile_test(100)
-# Pause profiling
+# do interesting things here
 IntelITT.pause()
+
+# convenience macro for the above:
+IntelITT.@collect begin
+    # ...
+end
 ```
 
-## Julia and VTunes
+## Trace instrumentation
+
+IntelITT.jl can also be used to enrich the trace data with custom information. For example,
+using tasks you can annotate a logical unit of work (e.g., a function call):
+
+```julia
+dom = Domain("MyApplication")
+task_begin(dom, "my task")
+# ...
+task_end(dom)
+
+# or, using a convenience macro:
+IntelITT.@task "my task" begin
+    # ...
+end
+```
 
-You need to set the environment variable `ENABLE_JITPROFILING=1`.
```

Comment:
This is still worth mentioning, right? Same for the `USE_INTEL_JITEVENTS=1` comment below for Julia < 1.9.